### PR TITLE
[reland] Add Material 3 support for `ListTile` - Part 1

### PR DESCRIPTION
**Pull request by TahaTesser**
reland https://github.com/flutter/flutter/pull/116194

With a regression test and updated `dense` documentation.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/material/list_tile.dart [hunk 1]
```diff
@@ -364,6 +368,8 @@ class ListTile extends StatelessWidget {
   /// If this property is null then its value is based on [ListTileTheme.dense].
   ///
   /// Dense list tiles default to a smaller height.
+  ///
+  /// When [ThemeData.useMaterial3] is true, [dense] wil not have any effect.
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/list_tile.dart:372]: This doesn't appear to be true. I think what we want to suggest is that we don't recommend that Material3 apps set dense to true.

Comment (TahaTesser) [reply packages/flutter/lib/src/material/list_tile.dart:372]: You're right. We're not skipping `dense` for M3.

**Review by HansMuller — APPROVED** [review]
LGTM

**Comment by HansMuller** [comment]
The "Google Testing" failure is due to a few golden image tests that already depend on `useMaterial3: true`. It should be possible to land this PR; coordinating that now.

**Comment by CaseyHillers** [comment]
FYI @itsjustkevin this has some trivial goldens that should be accepted. When testing comes through, can you verify there's no red assertion failures in the scuba?

**Pull request merged**